The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20525338160/job/58967595941) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/58967595941) [(plain)](https://github.com/rust-lang/rust/commit/ff71c5fcfe4d06169bb64070fba76763b9bee263/checks/58967595941/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
linting javascript files and applying suggestions
Running eslint on rustdoc JS files

/checkout/src/librustdoc/html/static/js/main.js
  2166:23  error  Strings must use doublequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

rerun tidy with `--extra-checks=js:lint --bless` to apply eslint suggestions
tidy [extra_checks]: checks with external tool 'eslint' failed
tidy [extra_checks]: FAIL
tidy: The following check failed: extra_checks
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 /node/bin/yarn --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1612:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1357:29

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Build completed unsuccessfully in 0:02:54
  local time: Fri Dec 26 16:10:50 UTC 2025
  network time: Fri, 26 Dec 2025 16:10:50 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/librustdoc/html/static/js/main.js
+++ b/src/librustdoc/html/static/js/main.js
@@ -2163,7 +2163,15 @@ function preLoadCss(cssUrl) {
             // Should never happen, but the world is a dark and dangerous place.
             return;
         }
-        copyContentToClipboard(codeElem.textContent);
+        let content = "";
+        for (const node of codeElem.childNodes) {
+            // We exclude line numbers.
+            if (node instanceof HTMLElement && node.hasAttribute("data-nosnippet")) {
+                continue;
+            }
+            content += node.textContent;
+        }
+        copyContentToClipboard(content);
     }
 
     /**

--- a/tests/rustdoc-gui/copy-code.goml
+++ b/tests/rustdoc-gui/copy-code.goml
@@ -42,7 +42,31 @@ store-size: (".example-wrap:nth-of-type(1) .copy-button", {
 })
 assert: |copy_height| > 0 && |copy_width| > 0
 
-// Checking same things for the copy button when there is no run button.
+
+// We now check that copying code when line numbers are displayed don't include these line numbers.
+
+// First we ensure that the clipboard is empty.
+assert-clipboard: ""
+// We make the line numbers appear.
+click: "rustdoc-toolbar .settings-menu"
+wait-for-css: ("#settings", {"display": "block"})
+// We make the line numbers appear.
+click: "#line-numbers"
+wait-for-local-storage: {"rustdoc-line-numbers": "true" }
+// We close the settings menu.
+click: "rustdoc-toolbar .settings-menu"
+wait-for-css-false: ("#settings", {"display": "block"})
+// We ensure that there are actually line numbers generated in the DOM.
+assert-text: (".example-wrap pre.rust code span[data-nosnippet]", "1")
+// We make the copy button appear.
+move-cursor-to: ".example-wrap pre.rust code"
+wait-for-css: (".example-wrap .button-holder", {"visibility": "visible"})
+// We click on the copy button.
+click: ".example-wrap .button-holder .copy-button"
+assert-clipboard: 'println!("nothing fancy");
+println!("but with two lines!");'
+
+// Back to UI checks, checking same things for the copy button when there is no run button.
 go-to: "file://" + |DOC_PATH| + "/lib2/sub_mod/struct.Foo.html"
 call-function: ("check-copy-button", {})
 // Ensure there is no run button.

